src/Controls/src/Core/BindableObject.cs

```diff
@@ -332,17 +333,12 @@ public static void SetInheritedBindingContext(BindableObject bindable, object va
 		{
 			//I wonder if we coulnd't treat bindingcoutext with specificities
 			BindablePropertyContext bpContext = bindable.GetContext(BindingContextProperty);
-			if (bpContext != null && bpContext.Values.LastOrDefault().Key.CompareTo(SetterSpecificity.ManualValueSetter) >= 0)
+			if (bpContext != null && bpContext.Values.GetSpecificityAndValue().Key.CompareTo(SetterSpecificity.ManualValueSetter) >= 0)
 				return;
 
-			object oldContext = bindable._inheritedContext?.Target;
-
-			if (ReferenceEquals(oldContext, value))
+			if (ReferenceEquals(bindable._inheritedContext?.Target, value))
 				return;
 
-			if (bpContext != null && oldContext == null)
-				oldContext = bpContext.Values.LastOrDefault().Value;
```

Comment:
I found `oldContext` wasn't used below, Roslyn actually recommended removing this.